src/vs/workbench/contrib/terminal/browser/terminalService.ts

```diff
@@ -305,6 +315,86 @@ export class TerminalService implements ITerminalService {
 
 		// Create async as the class depends on `this`
 		timeout(0).then(() => this._instantiationService.createInstance(TerminalEditorStyle, document.head));
+
+		//TODO:@meganrogge is there a better way to do this?
+		this.onDidReceiveProcessId(async () => await this._refreshDescriptionCwd());
```

Comment:
How about we have a new generic pty host message something like this to avoid all this piping through in the future for similar props?

```ts
onDidChangeProperty(name: TerminalProperty)

const enum TerminalProperty {
  cwd,
  initialCwd, // This wouldn't change but we could fetch it using a shared mechanism
  // We could add more like below after we validate this is a good idea
  // title,
  // icon,
}
```

These could always get synced over to the renderer when it changes so accesses are free?